src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -125,13 +224,25 @@ public override Task<DataTable> GetSchemaAsync(string collectionName, Cancellati
         public override Task<DataTable> GetSchemaAsync(string collectionName, string?[] restrictionValues, CancellationToken cancellationToken = default)
             => throw new NotImplementedException();
 
+        #region open
+
         /// <inheritdoc/>
-        public override void Open()
-            => throw new NotImplementedException();
+        public override void Open() => _Open(false, CancellationToken.None).GetAwaiter().GetResult();
 
         /// <inheritdoc/>
-        public override Task OpenAsync(CancellationToken cancellationToken)
-            => throw new NotImplementedException();
+        public override Task OpenAsync(CancellationToken cancellationToken) => _Open(true, cancellationToken);
+
+        private async Task _Open(bool async, CancellationToken cancellationToken)
```

Comment:
You can call it OpenInternal and CloseInternal for the internal close method instead, to keep naming conventions use PascalCase.